src/hotspot/share/interpreter/bytecodes.cpp

```diff
@@ -29,13 +29,309 @@
+  XX(_monitorenter    , "monitorenter"    , "b"    , nullptr    , T_VOID   , -1, true , _monitorenter   ) \
+  XX(_monitorexit     , "monitorexit"     , "b"    , nullptr    , T_VOID   , -1, true , _monitorexit    ) \
+  XX(_wide            , "wide"            , ""     , nullptr    , T_VOID   ,  0, false, _wide           ) \
+  XX(_multianewarray  , "multianewarray"  , "bkkc" , nullptr    , T_OBJECT ,  1, true , _multianewarray ) \
+  XX(_ifnull          , "ifnull"          , "boo"  , nullptr    , T_VOID   , -1, false, _ifnull         ) \
+  XX(_ifnonnull       , "ifnonnull"       , "boo"  , nullptr    , T_VOID   , -1, false, _ifnonnull      ) \
+  XX(_goto_w          , "goto_w"          , "boooo", nullptr    , T_VOID   ,  0, false, _goto_w         ) \
+  XX(_jsr_w           , "jsr_w"           , "boooo", nullptr    , T_INT    ,  0, false, _jsr_w          ) \
+  XX(_breakpoint      , "breakpoint"      , ""     , nullptr    , T_VOID   ,  0, true , _breakpoint     ) \
+  JVM_BYTECODES(XX)
+
+bool Bytecodes::_is_initialized = false;
+
+const char* const Bytecodes::_name[Bytecodes::number_of_codes] = {
+#define BYTECODE_NAME(code, name, format, wide_format, result_type, depth, can_trap, java_code) name,
+  BYTECODES(BYTECODE_NAME)
+#undef BYTECODE_NAME
+};
+
+const BasicType Bytecodes::_result_type[Bytecodes::number_of_codes] = {
+#define BYTECODE_RESULT_TYPE(code, name, format, wide_format, result_type, depth, can_trap, java_code) result_type,
+  BYTECODES(BYTECODE_RESULT_TYPE)
+#undef BYTECODE_RESULT_TYPE
+};
+
+const s_char Bytecodes::_depth[Bytecodes::number_of_codes] = {
+#define BYTECODE_DEPTH(code, name, format, wide_format, result_type, depth, can_trap, java_code) depth,
+  BYTECODES(BYTECODE_DEPTH)
+#undef BYTECODE_DEPTH
+};
+
+// Helper for determining the size (a.k.a. length) of a string literal.
+struct StringLiteralSize {
+  template <size_t N>
+  static constexpr size_t invoke(const char (&)[N]) {
+    static_assert(N > 0, "N must be greater than 0");
+    // The size is N - 1, as C strings have an implicit NUL at the end. So "foo" will result in N
+    // being 4, but we actually want 3.
+    return N - 1;
+  }
+
+  static constexpr size_t invoke(std::nullptr_t) {
+    return 0;
+  }
+};
+
+#define STRING_SIZE(string) StringLiteralSize::invoke(string)
+
+const u_char Bytecodes::_lengths[Bytecodes::number_of_codes] = {
+#define BYTECODE_LENGTHS(code, name, format, wide_format, result_type, depth, can_trap, java_code) (STRING_SIZE(wide_format) << 4) | (STRING_SIZE(format) & 0xf),
+  BYTECODES(BYTECODE_LENGTHS)
+#undef BYTECODE_LENGTHS
+};
+
+const Bytecodes::Code Bytecodes::_java_code[Bytecodes::number_of_codes] = {
+#define BYTECODE_JAVA_CODE(code, name, format, wide_format, result_type, depth, can_trap, java_code) Bytecodes::java_code,
+  BYTECODES(BYTECODE_JAVA_CODE)
+#undef BYTECODE_JAVA_CODE
+};
+
+jchar Bytecodes::_flags[(1<<BitsPerByte)*2];
```

Comment:
Why is this not _flags[Bytecodes::number_of_codes] ?

Reply:
It should probably be `Bytecodes::number_of_codes * 2`. Two entry per byte-code. One for the normal format and one for wide. Wasn't sure if it was safe or some code was relying on there being exactly 512 entries even if bytecode count is less than 256.

Reply:
When this was put in (JDK-6939207) the flags were arranged so that the two forms were at index N and N+256 rather than index N and N+number_of_codes. That seems to remain the case today - just grep for `1<<BitsPerByte`